iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -34,6 +34,49 @@ typedef NS_OPTIONS(NSUInteger, MWMLocationFrameworkUpdate) {
   BicycleRouting
 };
 
+static std::string DebugPrint(GeoMode geoMode) {
+  using enum GeoMode;
+  switch (geoMode) {
+    case Pending: return "Pending";
+    case InPosition: return "InPosition";
+    case NotInPosition: return "NotInPosition";
+    case FollowAndRotate: return "FollowAndRotate";
+    case VehicleRouting: return "VehicleRouting";
+    case PedestrianRouting: return "PedestrianRouting";
+    case BicycleRouting: return "BicycleRouting";
+  }
+}
+
+static std::string DebugPrint(MWMMyPositionMode mode) {
+  switch (mode) {
+    case MWMMyPositionModePendingPosition: return "MWMMyPositionModePendingPosition";
+    case MWMMyPositionModeNotFollowNoPosition: return "MWMMyPositionModeNotFollowNoPosition";
+    case MWMMyPositionModeNotFollow: return "MWMMyPositionModeNotFollow";
+    case MWMMyPositionModeFollow: return "MWMMyPositionModeFollow";
+    case MWMMyPositionModeFollowAndRotate: return "MWMMyPositionModeFollowAndRotate";
+  }
+}
+
+static std::string DebugPrint(MWMLocationStatus status) {
+  switch (status) {
+    case MWMLocationStatusNoError: return "MWMLocationStatusNoError";
+    case MWMLocationStatusNotSupported: return "MWMLocationStatusNotSupported";
+    case MWMLocationStatusDenied: return "MWMLocationStatusDenied";
+    case MWMLocationStatusGPSIsOff: return "MWMLocationStatusGPSIsOff";
+    case MWMLocationStatusTimeout: return "MWMLocationStatusTimeout";
+  }
+}
+
+static std::string DebugPrint(CLAuthorizationStatus status) {
+  switch (status) {
+    case kCLAuthorizationStatusNotDetermined: return "kCLAuthorizationStatusNotDetermined";
+    case kCLAuthorizationStatusRestricted: return "kCLAuthorizationStatusRestricted";
+    case kCLAuthorizationStatusDenied: return "kCLAuthorizationStatusDenied";
+    case kCLAuthorizationStatusAuthorizedAlways: return "kCLAuthorizationStatusAuthorizedAlways";
+    case kCLAuthorizationStatusAuthorizedWhenInUse: return "kCLAuthorizationStatusAuthorizedWhenInUse";
+  }
```

Comment:
Maybe put everywhere at the end `CHECK(false, ("Unsupported value", static_cast<int>(status)));` ?

Reply:
Fixed! Thanks!